src/components/List.jsx

```diff
@@ -0,0 +1,31 @@
+const List = ({ items, removeItem, editItem, markCompleted }) => {
+    return (
+        <>
+            <div className="flex flex-col gap-4">
+                {items.map((item) => {
+                    console.log(item.completed)
+                    const { id, title } = item
+                    return (
+                        <ul key={id} className="w-full">
+                            <li className="flex items-center gap-2 border-b py-4 w-full">
+                                <div className="w-1/2 flex gap-2 items-center">
+                                <input type="checkbox" onClick={() => markCompleted(id)} />
```

Comment:
It is better not to mix controlled input and uncontrolled input. You can the checked attribute on the checkbox to make it fully controlled.
```suggestion
                                <input type="checkbox" checked={item.completed} onClick={() => markCompleted(id)} />
```